src/arduino.cc/builder/ctags_target_file_saver.go

```diff
@@ -59,3 +67,53 @@ func (s *CTagsTargetFileSaver) Run(context map[string]interface{}) error {
 
 	return nil
 }
+
+func saveLinesContainingDirectivesAndSketch(src string, tofind []string) string {
+	lines := strings.Split(src, "\n")
+
+	saveLine := false
+	minimizedString := ""
+
+	for _, line := range lines {
+		if (saveLine || startsWithHashtag(line)) {
+			minimizedString += line + "\n"
+		}
+		if (containsAny(line, tofind)) {
```

Comment:
Why no `isLineDirective()` check here? Sounds like it could somehow trigger unintended matches when the sketch location somehow occurs in the included code somewhere?